h/services/search_index/_queue.py

```diff
@@ -127,12 +127,12 @@ def sync(self, limit):
           job on the queue to be re-checked and removed the next time the
           method runs.
         """
-        counts = Counter()
-
         jobs = self._get_jobs_from_queue(limit)
 
         if not jobs:
-            return counts
+            return {}
+
+        counts = defaultdict(set)
```

Comment:
The problem was that you can have more than one job for the same annotation. The annotation won't end up getting synced to Elasticsearch because the code uses a set for the list of `annotation_ids` to sync. But it _was_ getting double counted by the `Counter`. So use sets for counting too